apps/ecosystem/src/ecosystem.service.ts

```diff
@@ -419,7 +422,106 @@ export class EcosystemService {
     }
   }
 
-  async checkLedgerMatches(orgDetails: OrganizationData, ecosystemId: string): Promise<boolean> {
+
+  async addOrganizationsInEcosystem(ecosystemLeadOrgs: IEcosystemLeadOrgs): Promise<{
+    results: { statusCode: number; message: string; error?: string; data?: { orgId: string } }[];
+    statusCode: number;
+    message: string;
+  }> {
+    try {
+      const ecosystemRoleDetails = await this.ecosystemRepository.getEcosystemRole(EcosystemRoles.ECOSYSTEM_MEMBER);
+      const { organizationIds } = ecosystemLeadOrgs;
+      const results: { statusCode: number; message: string; error?: string; data?: { orgId: string } }[] = [];
+      const addedOrgs = [];
+      let successCount: number = 0;
+      let errorCount: number = 0;
+
+      for (const orgId of organizationIds) {
+        const checkOrgExists = await this.ecosystemRepository.checkOrgExists(orgId);
+        const result: { statusCode: number; message: string; error?: string; data?: { orgId: string } } = {
+          statusCode: 0,
+          message: ''
+        };
+
+        if (checkOrgExists) {
+          const orgAgentDetails = await this.ecosystemRepository.getAgentDetails(orgId);
+
+          if (orgAgentDetails?.orgDid) {
+            const existingOrg = await this.ecosystemRepository.checkOrgExistsInEcosystem(
+              orgId,
+              ecosystemLeadOrgs.ecosystemId
+            );
+            if (!existingOrg) {
+              addedOrgs.push({
+                orgId,
+                ecosystemId: ecosystemLeadOrgs.ecosystemId,
+                ecosystemRoleId: ecosystemRoleDetails.id,
+                status: EcosystemOrgStatus.ACTIVE,
+                deploymentMode: DeploymentModeType.PROVIDER_HOSTED,
+                createdBy: ecosystemLeadOrgs.userId,
+                lastChangedBy: ecosystemLeadOrgs.userId
+              });
+              successCount++;
+            } else {
+              result.statusCode = HttpStatus.CONFLICT;
+              result.message = `${ResponseMessages.ecosystem.error.orgAlreadyExists}`;
+              result.error = `${ResponseMessages.ecosystem.error.unableToAdd}`;
+              result.data = { orgId };
+              errorCount++;
+            }
+          } else {
+            result.statusCode = HttpStatus.BAD_REQUEST;
+            result.message = `${ResponseMessages.ecosystem.error.agentNotSpunUp}`;
+            result.error = `${ResponseMessages.ecosystem.error.unableToAdd}`;
+            result.data = { orgId };
+            errorCount++;
+          }
+        } else {
+          result.statusCode = HttpStatus.NOT_FOUND;
+          result.message = `${ResponseMessages.ecosystem.error.orgNotExist}`;
+          result.error = `${ResponseMessages.ecosystem.error.unableToAdd}`;
+          result.data = { orgId };
+          errorCount++;
+        }
+        if (0 !== result.statusCode) {
+          results.push(result);
+        }
+      }
+
+      const orgs = addedOrgs.map((item) => item.orgId);
```

Comment:
@bhavanakarwade Please add one more check to check the length of `addedOrgs` array length and throw error.